ObjectCreation/BSpline3D.FCMacro

```diff
@@ -0,0 +1,123 @@
+#*   published by the Free Software Foundation, either version 2.1 of the  *
+#*   License, or (at your option) any later version.                       *
+#*                                                                         *
+#*   FreeCAD is distributed in the hope that it will be useful, but        *
+#*   WITHOUT ANY WARRANTY; without even the implied warranty of            *
+#*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU      *
+#*   Lesser General Public License for more details.                       *
+#*                                                                         *
+#*   You should have received a copy of the GNU Lesser General Public      *
+#*   License along with FreeCAD. If not, see                               *
+#*   <https://www.gnu.org/licenses/>.                                      *
+#*                                                                         *
+#**************************************************************************/
+
+_Name__ = 'BSpline3D'
+__Comment__ = 'Create a parametric BSpline from Vertices selected in 3D view'
+__Author__ = 'CorvusCorax'
+__Version__ = '0.0.1'
+__Date__ = '2024-11-11'
+__License__ = 'LGPL-2.1-or-later'
+__Web__ = 'https://github.com/CorvusCorax/FreeCAD-BSpline3D'
+__Wiki__ = 'https://github.com/CorvusCorax/FreeCAD-BSpline3D/wiki'
+__Help__ = 'Select two or more vertices in the 3D view, then execute the macro.'
+__Status__ = 'beta'
+__Requires__ = 'FreeCAD >= v0.20'
+__Contact__ = 'https://github.com/CorvusCorax'
+__Communication__ = 'https://github.com/CorvusCorax/FreeCAD-BSpline3D/issues'
+__Files__ = ''
+__Xpm__ = """
+/* XPM */
+static char * XFACE[] = {
+"32 32 2 1",
+"  c #FFFFFF",
+". c #000000",
+"                                ",
+"                                ",
+"                                ",
+"                                ",
+"                                ",
+"              ........         .",
+"     ....   ...      ...    . . ",
+"    ..... ...          .   ...  ",
+"    .  ....            ..  .... ",
+"    .  ...              .  ...  ",
+"    .....               . .     ",
+"       ..                       ",
+"       .                        ",
+"      ..             ....       ",
+"      .             .....       ",
+"      ..            .  ..       ",
+"       .            .  ..       ",
+"       ..           ....        ",
+"        .                       ",
+"        ..                      ",
+"         ..                     ",
+"          ..                    ",
+"           ..                   ",
+"      .......                   ",
+"     ........                   ",
+"     .    ...                   ",
+"     .    ...                   ",
+"     .    ..                    ",
+"     ......                     ",
+"                                ",
+"                                ",
+"                                "};
+"""
+
+import FreeCAD, FreeCADGui
```

Comment:
Please use
`import FreeCAD as app` and `import FreeCADGui as gui` to be consistent with most of other macros.